WorkloadManagementSystem/Utilities/PilotCStoJSONSynchronizer.py

```diff
@@ -369,3 +380,20 @@ def _upload(self, pilotDict=None, filename='', pilotScript=''):
         except requests.ConnectionError:
           # if we are here it is probably because we are trying to upload to a WS that does not expose a POST API
           self.log.error("Can't issue POST", "on %s" % pfServer)
+
+  def _checksumFile(self, filePath):
+    """Calculate the checksum for the file and add to self._checkSumDict.
+
+    :param str filePath: path to the file for which to calculate the checksum
+    :returns None:
+    """
+    filename = os.path.basename(filePath)
+    self._checksumDict[filename] = hashlib.sha512(open(filePath, 'rb').read()).hexdigest()
+
+  def _syncChecksum(self):
+    """Upload the checksum file to the fileservers."""
+    with open('checksums.sha512', 'wb') as chksums:
```

Comment:
Why is this in binary mode? It will fail in Python 3 as you write `str` rather than `bytes`.